system-reinstall-bootc/src/users.rs

```diff
@@ -102,11 +102,58 @@ impl Display for UserKeys {
     }
 }
 
+#[derive(Debug)]
+struct SshdConfig {
+    authorized_keys_files: Vec<String>,
+    authorized_keys_command: String,
+    authorized_keys_command_user: String,
+}
+
+impl SshdConfig {
+    pub fn parse() -> Result<Self> {
+        let sshd_output = Command::new("sshd")
+            .arg("-T")
+            .run_get_string()
+            .context("running sshd -T")?;
+        tracing::trace!("sshd output:\n {}", sshd_output);
+
+        let mut config = SshdConfig {
+            authorized_keys_files: Vec::new(),
+            authorized_keys_command: String::new(),
+            authorized_keys_command_user: String::new(),
+        };
+
+        for line in sshd_output.lines() {
+            let mut kv = line.splitn(2, char::is_whitespace);
+            if let (Some(key), Some(value)) = (kv.next(), kv.next()) {
+                match key.trim() {
+                    "authorizedkeysfile" => {
```

Comment:
I wonder if it's maybe easier to just scrape the whole config into a `HashMap<&str, &str>` (note we don't need to copy the data!) and then to pick out specific values we want after that.

Something like this:

```
let config = sshd_output.lines()
        .filter_map(|l| l.split_once(' '))
        .collect::<BTreeMap<_, _>>();
let authorized_keys_file = config.get("authorizedkeysfile").unwrap_or("");
...
```

or so?